### ISSUE TITLE
Caret ranges don't handle 0.x versions properly

### ISSUE THREAD

**Issue by baptiste0928** [omichelsen/compare-versions#65]
Caret ranges (`^`) are defined as the following in the [npm semver documentation`](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004).

> Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple. In other words, this allows patch and minor updates for versions 1.0.0 and above, patch updates for versions 0.X >=0.1.0, and no updates for versions 0.0.X.

`compare-versions` does not follow this behavior and treat all `0.x.x` versions as compatible. Here is a simple example to reproduce the issue:

```js
import { satisfies } from 'compare-versions';

console.log(satisfies("1.2.3", "^1.1.2")); // true - ok
console.log(satisfies("0.3.4", "^0.2.3")); // true - should be false 
console.log(satisfies("0.0.4", "^0.0.3")); // true - should be false
```

**Comment by piotr-cz** [comment]
There is same issue for pre-release tags, and differs from [npm semantic versioner](https://semver.npmjs.com/)

```js
import { satisfies } from 'compare-versions';

console.log(satisfies('1.0.0-beta.3', '^1.0.0-beta.2')); // true - ok
console.log(satisfies('1.0.0-beta.3', '^1.0.0-beta.4')); // true - should be false
console.log(satisfies('1.0.0-beta.3', '~1.0.0-beta.4')); // true - should be false
```

**Comment by omichelsen** [comment]
Thanks for reporting this issue. Working on a fix.

**Comment by omichelsen** [comment]
Have a fix out now as v5.0.2, let me know if it works for you.

**Comment by piotr-cz** [comment]
Issue persists in v5.0.2 for both [0.x](https://runkit.com/embed/gwtt0qog4a5g) and [pre-releases](https://runkit.com/embed/wlhny6lr1mcq) with examples from issue description and my https://github.com/omichelsen/compare-versions/issues/65#issuecomment-1335187067.

Tested on https://npm.runkit.com/compare-versions - in both times for given 3 cases result was `true`.

**Comment by omichelsen** [comment]
I released an old version by mistake, 5.0.3 should be the correct fix version.

**Comment by piotr-cz** [comment]
It's fixed in 5.0.3, thank you!
